lib/membrane/video_compositor/server_runner.ex

```diff
@@ -101,15 +108,19 @@ defmodule Membrane.LiveCompositor.ServerRunner do
     end
   end
 
-  @spec wait_for_lc_startup(:inet.port_number(), pid()) :: :started | :not_started
-  defp wait_for_lc_startup(lc_port, pid) do
+  @spec wait_for_lc_startup(:inet.port_number(), pid(), String.t()) :: :started | :not_started
+  defp wait_for_lc_startup(lc_port, pid, instance_id) do
     0..300
     |> Enum.reduce_while(:not_started, fn _i, _acc ->
       Process.sleep(100)
 
       with {:is_alive, true} <- {:is_alive, Process.alive?(pid)},
-           {:ok, _} <- Request.get_status(lc_port) do
-        {:halt, :started}
+           {:ok, response} <- Request.get_status(lc_port) do
+        if response.body["instance_id"] == instance_id do
+          {:halt, :started}
+        else
+          {:halt, :not_started}
+        end
```

Comment:
Either log the error reason here or return the different value and log it in this function caller. The default error message wouldn't be helpful.